chore: Add `focus-trap` example

ItsJonQ opened the pull request:
This update adds a basic example for the `FocusTrap` component as part of this effort: https://github.com/reakit/reakit/issues/939

**Screenshots**

<img width="306" alt="FocusTrap example update" src="https://user-images.githubusercontent.com/2322354/152088386-696f4280-95b9-462e-a14e-b6d1c59d6638.png">

**How to test?**

- Run `yarn dev`
- Visit http://localhost:3000/examples/focus-trap.
- Press `Tab`.
- Toggle Focus Trap will trap `Tab` focus.

**Does this PR introduce breaking changes?**

Nope! Only adding an example.

diegohaz commented:
Great job! I'd suggest we make this example more simple though. And let users disable the focus trap behavior so keyboard users accessing the docs will be able to use the site. Let's also avoid using `ariakit-utils` in the examples.

Wrote this snippet to illustrate what I'm talking about:
```tsx
import { FocusEvent, useRef } from "react";
import { useCheckboxState, Checkbox, CheckboxCheck } from "ariakit/checkbox";
import { FocusTrap } from "ariakit/focus-trap";
import "./style.css";

export default function Example() {
  const focusTrapped = useCheckboxState({ defaultValue: true });
  const firstRef = useRef<HTMLButtonElement>(null);
  const lastRef = useRef<HTMLButtonElement>(null);

  const onTrapFocus = (event: FocusEvent) => {
    if (event.relatedTarget === firstRef.current) {
      lastRef.current?.focus();
    } else {
      firstRef.current?.focus();
    }
  };

  return (
    <>
      {focusTrapped.value && <FocusTrap onFocus={onTrapFocus} />}
      <div className="wrapper">
        <button className="button" ref={firstRef}>
          First
        </button>
        <Checkbox
          state={focusTrapped}
          clickOnEnter
          as="button"
          className="button"
        >
          <CheckboxCheck />
          Trap focus
        </Checkbox>
        <button className="button" ref={lastRef}>
          Last
        </button>
      </div>
      {focusTrapped.value && <FocusTrap onFocus={onTrapFocus} />}
    </>
  );
}
```

ItsJonQ commented:
@diegohaz Wow! Thank you so much 🙇 

I've updated the example based on your snippet. I adjusted the styles a bit to make the example a little clearer.

<img width="306" alt="FocusTrap example update" src="https://user-images.githubusercontent.com/2322354/152088337-9c0ca501-3839-46a2-bf4b-3e9c4bc28c1e.png">

Let me know if this works! Happy to make any updates necessary

diegohaz commented on packages/ariakit/src/focus-trap/__examples__/focus-trap/index.tsx:
```diff
@@ -0,0 +1,37 @@
+  return (
+    <>
+      {focusTrapped.value && <FocusTrap onFocus={onTrapFocus} />}
+      <div className="wrapper">
+        <label className="label">
+          <Checkbox state={focusTrapped} clickOnEnter className="checkbox" />
+          Trap focus
+        </label>
+        <button className="button" ref={firstRef}>
+          First
+        </button>
+        <button className="button" ref={lastRef}>
+          Last
+        </button>
+      </div>
```
The behavior is a little bit weird now when the focus is trapped. When pressing Tab, the focus never goes to the checkbox, when pressing Shift+Tab, the focus never goes to the last button.

I think we can just make the checkbox the first item ref to fix this:
```suggestion
      <div className="wrapper">
        <label className="label">
          <Checkbox state={focusTrapped} ref={firstRef} className="checkbox" />
          Trap focus
        </label>
        <button className="button" ref={lastRef}>
          Button
        </button>
      </div>
```

ItsJonQ replied:
Thank you so much @diegohaz !! Sorry for the delay! I was able to update the example with your suggestion. I've also added the focus trap test. The Ariakit test utils are really great! ❤️

diegohaz reviewed (commented):
Niiice! I made a suggestion to fix the tab order.

Also, would you like to add some tests? We can test with `press.Tab()` and `press.ShiftTab()` (do a global search on the project to find some examples).

diegohaz commented on packages/ariakit/src/focus-trap/__examples__/focus-trap/test.tsx:
```diff
@@ -0,0 +1,46 @@
+test("correctly traps focus", async () => {
+  // Focus trap is on by default in the Example
+  render(
+    <div>
+      <Example />
+      <div tabIndex={0}>Outside</div>
+    </div>
```
Nice! 👍  We usually render just `<Example />`, but these extra elements are handy here. Do you think we should add another element before so we can test shift+tab as well?

ItsJonQ replied:
Oh yea! Let's do that 💪

diegohaz commented on packages/ariakit/src/focus-trap/__examples__/focus-trap/test.tsx:
```diff
@@ -0,0 +1,46 @@
+      <div tabIndex={0}>Outside</div>
+    </div>
+  );
+  focus(getByRole("checkbox"));
```
I think we should use something that better describes a user action here, like `await press.Tab()` or `await click()`.

diegohaz commented on packages/ariakit/src/focus-trap/__examples__/focus-trap/test.tsx:
```diff
@@ -0,0 +1,46 @@
+  );
+  // Disabling focus trap
+  await click(getByLabelText("Trap focus"));
+  await focus(getByRole("button"));
```
```suggestion
  await press.Tab();
```

ItsJonQ replied:
@diegohaz I updated both tests to rely on user interactions - `Tab` and `ShiftTab`. These utils are great!

diegohaz reviewed (approved):
Great work! Thank you! ❤️
